src/main/java/com/auth0/client/mgmt/ManagementAPI.java

```diff
@@ -41,7 +45,53 @@ public ManagementAPI(String domain, String apiToken) {
         telemetry = new TelemetryInterceptor();
         logging = new HttpLoggingInterceptor();
         logging.setLevel(Level.NONE);
-        client = new OkHttpClient.Builder()
+        client = buildNetworkingClient(options);
+    }
+
+    /**
+     * Create an instance with the given tenant's domain and API token.
+     * See the Management API section in the readme or visit https://auth0.com/docs/api/management/v2/tokens to learn how to obtain a token.
+     *
+     * @param domain   the tenant's domain.
+     * @param apiToken the token to authenticate the calls with.
+     */
+    public ManagementAPI(String domain, String apiToken) {
+        this(domain, apiToken, new ClientOptions());
+    }
+
+    /**
+     * Given a set of options, it creates a new instance of the {@link OkHttpClient}
+     * configuring them according to their availability.
+     *
+     * @param options the options to set to the client.
+     * @return a new networking client instance configured as requested.
+     */
+    private OkHttpClient buildNetworkingClient(ClientOptions options) {
+        OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
+        final ProxyOptions proxyOptions = options.getProxyOptions();
+        if (proxyOptions != null) {
+            //Set proxy
+            clientBuilder.proxy(proxyOptions.getProxy());
+            //Set authentication, if present
+            final String proxyAuth = proxyOptions.getBasicAuthentication();
+            if (proxyAuth != null) {
+                clientBuilder.proxyAuthenticator(new Authenticator() {
+
+                    private static final String PROXY_AUTHORIZATION_HEADER = "Proxy-Authorization";
+
+                    @Override
+                    public okhttp3.Request authenticate(Route route, Response response) throws IOException {
+                        if (response.request().header(PROXY_AUTHORIZATION_HEADER) != null) {
+                            return null;
+                        }
+                        return response.request().newBuilder()
+                                .header(PROXY_AUTHORIZATION_HEADER, proxyAuth)
+                                .build();
+                    }
+                });
+            }
+        }
```

Comment:
I know this is duplicated code from `AuthAPI` because they are in different packages and we don't want to expose any public API dependencies on `OkHttpClient`. If there's any way we can at least get _some_ code reuse without leaking dependencies that would be ideal, but may not be possible. I'm sure you've already thought about this, so if no new idea comes to you, feel free to ignore this comment.

Reply:
I went through this but like you said, the project structure is already separates the 2 APIs. So, while the introduced options would be common to both API clients, there's no common package for me to hide this method there. So the options are either repeating the code (as seen here) or exposing the OkHttpClient in the signature, which we want to avoid 👍